server/src/test/java/org/elasticsearch/action/DocWriteResponseTests.java

```diff
@@ -107,4 +107,34 @@ public void testToXContentDoesntIncludeForcedRefreshUnlessForced() throws IOExce
             }
         }
     }
+
+    public void testTypeWhenCompatible() throws IOException {
```

Comment:
Is it possible to move this test (and any other compatible tests) over to the compatible module under the version7 package ?  Ideally, we keep this coverage, but make it super easy to remove.

Reply:
good idea, I think the v7 subdirectory in rest-compatibility and tweaking gradle testClasses should be enough. 
so that we can have the same packages with same classes but in different submodules (the situation when v7 is not yet removed, but more compatible code for v8 is being created)
otherwise suffixing test classes with v7?

Reply:
sounds good.
Since it mostly cosmetic, lets remove the v7 package, and only using the v7 in the class name. That should be sufficient to make it obvious it is v7 code and still allow us use package scope without creating new sub modules. We re-visit later if turns out too ugly :)

Reply:
I only was referring to tests - that if we have a copy of a class from a different module we could sufix with v7.
it feels right to have a version7 package in main though. as on the main `org.elasticsearch.rest.compat` level we could keep version agnostic classes - like RestCompatPlugin.